src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.sscs.util;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CcdValue;
+import uk.gov.hmcts.reform.sscs.ccd.domain.OtherParty;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.YesNo;
+
+public class OtherPartyDataUtil {
+
+    private OtherPartyDataUtil() {
+    }
+
+    public static void updateOtherPartyUcb(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getOtherParties() != null) {
+            sscsCaseData.setOtherPartyUcb(sscsCaseData.getOtherParties().stream()
+                    .filter(o -> YesNo.isYes(o.getValue().getUnacceptableCustomerBehaviour()))
+                    .map(o -> o.getValue().getUnacceptableCustomerBehaviour().getValue())
+                    .findAny().orElse(YesNo.NO.getValue()));
```

Comment:
Use static imports as much as possible, for readability
Move the .orElse into a new line for the same reason.

Reply:
Done